open-cluster-management/backlog#11989 depl sub

cadawson opened the pull request.

cadawson committed:
open-cluster-management/backlog#11989 depl sub

cadawson commented:
This will have some additional changes, so it is on hold.

cadawson committed:
open-cluster-management/backlog#11989 Sub dep ui

cadawson committed:
Merge branch '2.3_stage' into dawson1198923

cadawson committed:
open-cluster-management/backlog#11989 Upd sub

cadawson committed:
Merge branch 'dawson1198923' of github.com:open-cluster-management/rhacm-docs into dawson1198923

cadawson commented:
Hello @skeeey . The updates are in to this. Will you review again? Thank you!

skeeey commented on services/submariner.adoc:
```diff
@@ -470,10 +160,15 @@ $ az network nsg rule create --resource-group <res-group> \
++
 Replace `name` with your rule name.
 
+. Label a worker node in the cluster with the following label: `submariner.io/gateway=true`.
```
It's better to put this step at the end, after the step `Create the NSG rules to open 8080/TCP port  ...`

skeeey commented:
@cadawson  for table `Table 1. Submariner required ports`, we need correct it to

Name | Default value | Customizable
-- | -- | --
IPSec IKE | 500/UDP | Yes
IPSec NATT | 4500/UDP | Yes
VXLAN| 4800/UDP | No
Submariner Metrics Port| 8080/UDP | No

the `VXLAN` is not only for vsphere, all platforms need it

cadawson committed:
open-cluster-management/backlog#11989 nw sub deploy

cadawson committed:
open-cluster-management/backlog#11989 rev

cadawson commented:
I updated the content here @skeeey. I tried to merge the content from issues #13065 and #11989. What I was not sure of is should we remove the prep for the other platforms or leave them? I added the prep for the three platforms in the new Deploying with the APIs topic. We can take a look. Here is the PR: https://github.com/open-cluster-management/rhacm-docs/pull/2140. Thanks!

skeeey commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,123 @@
-. Create a `ManagedClusterSet` on the hub cluster by using the instructions provided in link:../clusters/custom_resource.adoc#managedclustersets[ManagedClusterSets]. The `submariner-addon` creates a namespace called `submariner-clusterset-<clusterset-name>-broker` and deploys the Submariner Broker to it. The name of the ManagedClusterSet replaces <clusterset-name> in the namespace name. Your entr ...
-+ 
-[source,yaml]
+. If you did not create the managed cluster with {product-title-short}, you must manually create a secret on your hub cluster in the namespace of your managed cluster that contains your AWS credential secret. You gather your credential information for the provider where you are deploying Submariner. If you created the cluster with {product-title-short}, then skip to step 3.
```
Please delete this step, for console way, we don't need this step, UI can distinguish this automatically, then UI will show different configuration pages

cadawson replied:
Step removed.

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+This configuration automatically opens the Submariner required ports: Internet Key Exchange (IKE) port (500/UDP), network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your AWS instance. It also creates one AWS instance as the Submariner gateway with the AWS instance type `m5n.large`.
++
+If you want to customize your Internet Key Exchange (IKE) port or network address translation - traversal (NATT) port, enter a command that contains information that is similar to the following example:
```
we may need a list number here `3. If you want to customize your Internet Key Exchange (....`

cadawson replied:
Added the number. This raises the question: Can these steps all be done in that order or should only one of the steps be done?

skeeey replied:
@cadawson, good point, these steps are independent each other, we may not use a number list, just use a list

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+Replace `managed-cluster-namespace` with the namespace of your managed cluster.
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
+
```
We may need an explanation

```
This configuration will use default Internet Key Exchange (IKE) port (500/UDP) and default network address translation - traversal (NATT) port (4500/UDP) for your Submariner and one worker node will be labled as the Submariner gateway in your vSphere cluster.
```

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
+
+. Submariner uses IP security (IPSec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPSec ports, or you can specify other IPSec ports that you configured. When you run this procedure without specifying an IPSec IKE port or an IPSec NATT port, the default ports are automatically used for the communication. The default ports are 500/UDP an ...
```
This is to use the customized IPsec ports, we may change this to

3. If you want to use your customized IPSec ports, enter a command that contains information that is similar to the following example:

```
cat << EOF | oc apply -f -
apiVersion: submarineraddon.open-cluster-management.io/v1alpha1
kind: SubmarinerConfig
metadata:
    name: submariner
    namespace: <managed-cluster-namespace>
spec:
    IPSecIKEPort: <IKEPort>
    IPSecNATTPort: <NATTPort>
EOF
```

Replace managed-cluster-namespace with the namespace of your managed cluster.

Replace IKEPort with the IKE port that you want to use.

Replace NATTPort with the NATT port that you want to use.

> Note: The name of the SubmarinerConfig must be submariner, as shown in the example.

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+
+To prepare VMware vSphere clusters for deploying Submariner, complete the following steps:
+
+. Ensure that the IPSec IKE, IPSec NATT, and VXLAN ports are open.
```
Ensure that the IPSec IKE, IPSec NATT, VXLAN and metrics ports are open.

skeeey commented:
@cadawson thank you, I add a few comments

cadawson committed:
open-cluster-management/backlog#11989 Rev

cadawson committed:
open-cluster-management/backlog#11989 rev

cadawson commented:
Added Sho as a reviewer, as he will be interested in the UI instructions.

cadawson commented:
I updated for your comments, @skeeey. Take a look, when you have a chance. Thank you!

showeimer commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,59 @@
++
+* `Agent status` indicates whether Submariner is successfully deployed on the managed cluster. The console might report a status of `Degraded` until it is installed and configured. 
++
+* `Gateway nodes labeled` indicates how many worker nodes are labeled with the Submariner gateway label: `submariner.io/gateway=true` on the managed cluster.
```
Seems good, there's kind of a problem after you install though where the statuses start showing Degraded even though the controller is still working on the installation.  Not sure if that should be a known issue @skeeey ?

skeeey replied:
@showeimer,  I think this is not a known issue, for submariner-addon, it cannot distinguish that the submariner is installing or the submariner is failed to install, it just know the current submariner status, e.g. if the submariner deployment cannot be found, we cannot distinguish it due to the submariner operator does not start to deploy submariner yet, or the submariner operator is deploying the submariner but the deployment is not created yet, or the submariner is deployed, but the deployment is deleted somehow, we can just know the current submariner status and say it is degrade or healthy

skeeey commented:
Invite Nir to help us to review

nyechiel commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,59 @@
+* `Instance type` - The Amazon Web Services EC2 instance type of the gateway node that is created on the managed cluster. The default value is `m5n.large`. This field is only visible when your managed cluster environment is AWS.
+* `IPsec IKE port` - The Submariner creates the IPsec tunnel between the clusters. This port allows Internet Key Exchange (IKE) protocol traffic on the Submariner gateway node to set up the IPsec tunnel. The default value is port `500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
+* `IPsec NAT-T port` - The Submariner creates the IPsec tunnel between the clusters.  This port is used for IPsec NAT traversal. The default value is port `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
+* `Gateway count` - The number of worker nodes that are used to deploy the Submariner gateway component on your managed cluster. The default value is `1`. If the value is greater than 1, the Submariner gateway HA is automatically enabled.
```
```suggestion
* `Gateway count` - The number of worker nodes that are used to deploy the Submariner gateway component on your managed cluster. The default value is `1`. If the value is greater than 1, Submariner gateway High Availability (HA) is automatically enabled.
```

nyechiel commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,59 @@
+* `IPsec IKE port` - The Submariner creates the IPsec tunnel between the clusters. This port allows Internet Key Exchange (IKE) protocol traffic on the Submariner gateway node to set up the IPsec tunnel. The default value is port `500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
+* `IPsec NAT-T port` - The Submariner creates the IPsec tunnel between the clusters.  This port is used for IPsec NAT traversal. The default value is port `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
+* `Gateway count` - The number of worker nodes that are used to deploy the Submariner gateway component on your managed cluster. The default value is `1`. If the value is greater than 1, the Submariner gateway HA is automatically enabled.
+* `Cable driver` - The Submariner gateway cable driver.
```
```suggestion
* `Cable driver` - The Submariner cable engine component that maintains the cross-cluster tunnels. Default is Libreswan IPsec.
```

nyechiel commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,59 @@
+* `AWS Secret Access Key` - This field is only visible when you import an AWS cluster.
+* `Google Cloud Platform service account JSON key` - This field is only visible when you import a Google Cloud Platform cluster.
+* `Instance type` - The Amazon Web Services EC2 instance type of the gateway node that is created on the managed cluster. The default value is `m5n.large`. This field is only visible when your managed cluster environment is AWS.
+* `IPsec IKE port` - The Submariner creates the IPsec tunnel between the clusters. This port allows Internet Key Exchange (IKE) protocol traffic on the Submariner gateway node to set up the IPsec tunnel. The default value is port `500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
```
```suggestion
* `IPsec IKE port` - The Internet Key Exchange (IKE) protocol traffic on the Submariner gateway node to set up the IPsec tunnel. The default value is `500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
```

skeeey replied:
We need remove this sentence, because the IKE port is not needed anymore

nyechiel commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,59 @@
+* `Google Cloud Platform service account JSON key` - This field is only visible when you import a Google Cloud Platform cluster.
+* `Instance type` - The Amazon Web Services EC2 instance type of the gateway node that is created on the managed cluster. The default value is `m5n.large`. This field is only visible when your managed cluster environment is AWS.
+* `IPsec IKE port` - The Submariner creates the IPsec tunnel between the clusters. This port allows Internet Key Exchange (IKE) protocol traffic on the Submariner gateway node to set up the IPsec tunnel. The default value is port `500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
+* `IPsec NAT-T port` - The Submariner creates the IPsec tunnel between the clusters.  This port is used for IPsec NAT traversal. The default value is port `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
```
```suggestion
* `IPsec NAT-T port` - The IPsec NAT traversal port. The default value is `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
```

nyechiel commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+
+To prepare VMware vSphere clusters for deploying Submariner, complete the following steps:
+
+. Ensure that the IPSec IKE, IPSec NATT, VXLAN, and metrics ports are open.
```
```suggestion
. Ensure that the IPsec IKE, IPsec NATT, VXLAN, and metrics ports are open.
```

skeeey replied:
we also need remove IKE port from here

cadawson replied:
IBM style

nyechiel commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+This configuration uses default Internet Key Exchange (IKE) port (500/UDP) and default network address translation - traversal (NATT) port (4500/UDP) for your Submariner and one worker node is labled as the Submariner gateway on your vSphere cluster.
++
+Submariner uses IP security (IPSec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPSec ports, or you can specify other IPSec ports that you configured. When you run this procedure without specifying an IPSec IKE port or an IPSec NATT port, the default ports are automatically used for the communication. The default ports are 500/UDP and  ...
```
```suggestion
Submariner uses IPsec to establish secure tunnels between the clusters on the gateway nodes. You can either use the default IPsec ports, or you can specify custom ports. When you run this procedure without specifying an IPsec IKE port or an IPsec NATT port, the default ports are automatically used. The default ports are 500/UDP and 4500/UDP, respectively. 
```

skeeey replied:
we also need remove IKE port from here

nyechiel commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+[#preparing-vm]
+=== Preparing to deploy Submariner on VMware vSphere
+
+Submariner uses IP security (IPSec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPSec ports, or you can specify other IPSec ports that you configured. When you run this procedure without specifying an IPSec IKE port or an IPSec NATT port, the default ports are automatically used for the communication. The default ports are 500/UDP and  ...
```
```suggestion
Submariner uses IPsec to establish secure tunnels between the clusters on the gateway nodes. You can either use the default IPsec ports, or specify custom ports. When you run this procedure without specifying an IPsec IKE port or an IPsec NATT port, the default ports are automatically used. The default ports are 500/UDP and 4500/UDP, respectively. 
```

skeeey replied:
we also need remove IKE port from here

nyechiel commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+|===
+| Name | Default value | Customizable 
+
+| IPSec IKE
```
```suggestion
| IPsec IKE
```

skeeey replied:
we also need remove IKE port from here

nyechiel commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+| 500/UDP
+| Yes
+
+| IPSec NATT
```
```suggestion
| IPsec NATT
```

cadawson replied:
IBM style is to use IPSec. I see where it is also used as IPsec, but we are now guided by IBM style for consistency.

nyechiel replied:
Thanks @cadawson. FYI, we use "IPsec" throughout the upstream Submariner docs. This is how the RFC refers to it   (https://datatracker.ietf.org/doc/html/rfc6071) and this is what the rest of the Red Hat docs use as well (for e.g https://docs.openshift.com/container-platform/4.7/networking/ovn_kubernetes_network_provider/about-ipsec-ovn.html or https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/securing_networks/configuring-a-vpn-with-ipsec_securing-networks).

cadawson replied:
Hmmm. Thanks for the information @nyechiel . Still learning that things are not as set in stone at Red Hat. Given your example, and that IPsec is mainly limited to this section, we will use IPsec. If anyone says anything, we might have to change it at some point. We should be ok, though. You make a good argument.

nyechiel commented:
@skeeey @showeimer sorry for catching this so late, but with Submariner 0.9/ACM 2.3 IKE port is no longer being used. The `deprecated` field is set in the Operator for backwards compatibility, but instead we keep all IPsec protocols in one single port (the one defined under NAT-T). We need to update the docs accordingly, as well as the UI. /cc @SteveMattar @mangelajo

SteveMattar commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+After the `ManagedClusterSet` is created, the `submariner-addon` creates a namespace called `<managed-cluster-set-name>-broker` and deploys the Submariner broker to it.
+
+. Add one managed cluster to the `ManagedClusterSet` by entering the following command:
++
```
Is the `ManagedClusterSetBinding` necessary here?

```
apiVersion: cluster.open-cluster-management.io/v1alpha1
kind: ManagedClusterSetBinding
metadata:
  name: submariner
  namespace: submariner-operator
spec:
  clusterSet: submariner
 ```

skeeey replied:
we don't need `ManagedClusterSetBinding` here

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
++
+This configuration automatically opens the Submariner required ports: Internet Key Exchange (IKE) port (500/UDP), network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your AWS instance. It also creates one AWS instance as the Submariner gateway with the AWS instance type `m5n.large`.
```
remove IKE port

This configuration automatically opens the Submariner required ports: network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your AWS instance. It also creates one AWS instance as the Submariner gateway with the AWS instance type `m5n.large`.

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+spec:
+    credentialsSecret:
+      name: <managed-cluster-name>-aws-creds
+    IPSecIKEPort: <IKEPort>
```
remove IKE port

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+This configuration automatically opens the Submariner required ports: Internet Key Exchange (IKE) port (500/UDP), network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your AWS instance. It also creates one AWS instance as the Submariner gateway with the AWS instance type `m5n.large`.
++
+. If you want to customize your Internet Key Exchange (IKE) port or network address translation - traversal (NATT) port, enter a command that contains information that is similar to the following example:
```
remove IKE port

If you want to customize your network address translation - traversal (NATT) port, enter a command that contains information that is similar to the following example:

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+Replace `managed-cluster-name` with the name of your managed cluster. The value of `managed-cluster-name-aws-creds` is your AWS credential secret name, which you can find in the cluster namespace of your hub cluster. 
++
+Replace `IKEPort` with the IKE port that you want to use.
```
remove IKE port

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
++
+This configuration automatically opens the Submariner required ports: Internet Key Exchange (IKE) port (500/UDP), network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your Google Cloud Platform instance. It also labels one worker node as the Submariner gateway in your Google Cloud Platform cluster ...
```
remove IKE port

This configuration automatically opens the Submariner required ports: network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your Google Cloud Platform instance. It also labels one worker node as the Submariner gateway in your Google Cloud Platform cluster.

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+spec:
+    credentialsSecret:
+      name: <managed-cluster-name>-gcp-creds
+    IPSecIKEPort: <IKEPort>
```
remove IKE port

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
++
+This configuration uses default Internet Key Exchange (IKE) port (500/UDP) and default network address translation - traversal (NATT) port (4500/UDP) for your Submariner and one worker node is labled as the Submariner gateway on your vSphere cluster.
```
we also need remove IKE port from here

This configuration uses default network address translation - traversal (NATT) port (4500/UDP) for your Submariner and one worker node is labled as the Submariner gateway on your vSphere cluster.

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
++
+Replace `managed-cluster-namespace` with the namespace of your managed cluster.
++
+Replace `IKEPort` with the IKE port that you want to use.
```
remove IKE port

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,408 @@
+    name: submariner
+    namespace: <managed-cluster-namespace>
+spec:
+    IPSecIKEPort: <IKEPort>
```
remove IKE port

skeeey commented on services/submariner.adoc:
```diff
@@ -54,337 +51,21 @@ Ensure that you have the following prerequisites before using Submariner:
+
+| Submariner metrics port
+| 8080/UDP
+| No
```
We need remove IKE port from this table, and update
> Firewall configuration open for 4500/UDP, 500/UDP, and any other ports that are used for IPSec traffic on the gateway nodes.

to 

> Firewall configuration open for 4500/UDP and any other ports that are used for IPSec traffic on the gateway nodes.

skeeey commented:
@showeimer I logged an issue to remove the ike port from gui https://github.com/open-cluster-management/backlog/issues/13902, please take a look, thanks

cadawson committed:
open-cluster-management/backlog#11989 rev

cadawson committed:
open-cluster-management/backlog#11989 rev

cadawson committed:
open-cluster-management/backlog#11989 Rev

cadawson committed:
open-cluster-management/backlog#11989 rev

cadawson committed:
open-cluster-management/backlog#11989 rev

cadawson committed:
open-cluster-management/backlog#11989 Rev

cadawson committed:
open-cluster-management/backlog#11989 rev list

cadawson commented:
Hello @skeeey . I removed the IKE port. If all is well, we can move forward., Thank you!

nyechiel commented:
@mangelajo @sridhargaddam @SteveMattar can you please review this?

SteveMattar commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,394 @@
+After the `ManagedClusterAddOn` is created, the `submariner-addon` deploys Submariner to the `submariner-operator` namespace on the managed cluster. You can view the deployment status of Submariner from the status of this `ManagedClusterAddOn`.
++
+*Note:* The name of `ManagedClusterAddOn` must be `submariner`.
+
```
Is this label added automatically after creating the `ManagedClusterAddon`?
`oc label managedclusters <managed-cluster-name> "cluster.open-cluster-management.io/submariner-agent=true>" --overwrite`

skeeey replied:
The label `cluster.open-cluster-management.io/submariner-agent` does not need any more in ACM 2.3

skeeey commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,394 @@
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
++
+This configuration automatically opens the Submariner required ports: network address translation - traversal (NATT) port (4500/UDP), virtual extensible LAN (VXLAN) port (4800/UCP), and Submariner metrics port (8080/TCP) on your Google Cloud Platform instance. It also labels one worker node as the Submariner gateway in your Google Cloud Platform cluster.
```
@cadawson we need a small update here

> It also labels one worker node as the Submariner gateway in your Google Cloud Platform cluster.

to

> It also labels one worker node as the Submariner gateway and enables the public IP of this node in your Google Cloud Platform cluster.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,360 +35,40 @@ Ensure that you have the following prerequisites before using Submariner:
 * IP connectivity must be configured between the Gateway nodes. When connecting two clusters, at least one of the clusters must have a publicly routable IP address designated to the Gateway node.
```
@cadawson here, we need update

> When connecting two clusters, at least one of the clusters must have a publicly routable IP address designated to the Gateway node.

to
 
> When connecting two clusters, at least one of the clusters must have a reachable either on its public or private IP address designated to the Gateway node, for more details, see [Submariner Nat Traversal](https://submariner.io/operations/nat-traversal)

skeeey commented:
@cadawson there are two small updates

https://github.com/open-cluster-management/rhacm-docs/pull/2140/files#r665906250

and 

https://github.com/open-cluster-management/rhacm-docs/pull/2140/files#r665912341

Thank you!

cadawson committed:
open-cluster-management/backlog#11989 Rev

cadawson commented:
@skeeey and @nyechiel : I updated the two items and replaced IPSec with IPsec in all places except in the examples. I wasn't sure if the examples would still work with "IPsec" instead of "IPSec". If it is not case sensitive, we can change those, too. Thanks!

skeeey commented:
/lgtm

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
-  name: <ManagedClusterSet-name>
-----
-Replace _ManagedClusterSet-name_ with a name for the `ManagedClusterSet` that you are creating.
+. On the Clusters page, select the *Cluster sets* tab. The clusters that you want enable with Submariner must be in the same cluster set. 
```
update to read: _Clusters_ page 

Also italicize Cluster set

cadawson replied:
Added italics to Clusters page. Not sure why we would add it to cluster set. Can you explain your thinking on that?

dockerymick replied:
It's the name of a tab, so it should be italicized

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
-----
-+
-Replace `_managedcluster-name_` with the name of the managed cluster that you want to use with Submariner. 
+. If the clusters where you want to deploy Submariner are already in the same cluster set, skip to step 5 to deploy Submariner.
```
If the clusters that you want to deploy to Submariner are already....

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
-Replace `_managedcluster-name_` with the name of the managed cluster that you want to use with Submariner. 
+. If the clusters where you want to deploy Submariner are already in the same cluster set, skip to step 5 to deploy Submariner.
+
+. If the clusters on which you want to deploy Submariner are not in the same cluster set, create a cluster set for them by completing the following steps: 
```
If the clusters that you want to deploy to Submariner are not...

cadawson replied:
We are deploying Submariner on the cluster, so I think this is ok. I did change the earlier one, because, as you pointed out, it read a little goofy.

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
+.. Select *Save* to save the cluster set, and view the resulting cluster set page.
 
-. Add the managed clusters to the `ManagedClusterSet` by entering the following command:
+. On the cluster set page, select the *Submariner add-ons* tab.
```
italicize Submariner add-ons

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
+
+. Select the clusters on which you want to deploy Submariner. 
+
+. Enter the following information in the *Install Submariner add-ons* wizard:
```
Italicize Install Submariner add-ons

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
+
+. Select *Install Submariner add-ons*.
+
+. Select the clusters on which you want to deploy Submariner. 
```
Select the clusters that you want to deploy to Submariner.

cadawson replied:
Deploying Submariner on the clusters, so I think this is ok.

dockerymick commented on services/deploy_submariner.adoc:
```diff
@@ -1,39 +1,58 @@
-. Repeat those steps for all of the managed clusters that you want to add to the `ManagedClusterSet`.
+. Click *Install* to deploy Submariner on the selected managed clusters. 
++
+It might take several minutes for the installation and configuration to complete. You can check the Submariner status in the list on the *Submariner add-ons* tab.
```
Italicize Submariner add-ons and use a colon (:) instead of a period since you are prompting the user.

cadawson committed:
open-cluster-management/backlog#11989 format

dockerymick commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,394 @@
+[#deploying-submariner-apis]
+= Deploying Submariner with APIs
+
+Before you deploy Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. Currently you can use the `SubmarinerConfig` API to automatically prepare the clusters on Amazon Web Services, Google Cloud Platform and VMware vSphere. For other platforms, you need to prepare them manually, see xref:../services/deploy_submariner_api.adoc#preparing-the- ...
```
I would add a comma after Currently

cadawson replied:
Agreed. Good catch.

dockerymick commented on services/deploy_submariner_api.adoc:
```diff
@@ -0,0 +1,394 @@
+
+* `SubmarinerGatewayNodesLabeled` condition indicates whether there are labeled Submariner gateway nodes on the managed cluster.
+* `SubmarinerAgentDegraded` condition indicates whether the Submariner is successfully deployed on the managed cluster.
+* `SubmarinerConnectionDegraded` condition indicates how many connections are established on the managed cluster with Submariner.
```
I'm not sure how you want to align the following statement: "After Submariner is deployed on the managed cluster, you can verify the Submariner deployment status by checking the status of submariner `ManagedClusterAddOn` by entering the following command:" 

Right now it is not aligned with the step.

This file is organized very well Chris; it's consistent, and easy to follow the flow considering the different scenarios the user has !

cadawson replied:
I had that one hanging out there outside of the procedure, but apparently it was not the right place. Based on your question, it kind of does look like another step, so I incorporated it into the procedure as a final step. 

Thanks for the feedback, and the kind words. This one has changed several times along the way, so I'm glad to hear that it turned out ok. Thank you for the review!

dockerymick reviewed (approved):
Made a few comments, awesome job on this Chris it looks really good

dockerymick commented:
Quick comment, remember to add the What's new entry for this topic if you havent added it already

cadawson committed:
open-cluster-management/backlog#11989 peer

cadawson committed:
open-cluster-management/backlog#11989 peer

cadawson commented:
@showeimer I know your role has changed. Did you want to take a look at this one, or are you ok? Thanks!

cadawson committed:
open-cluster-management/backlog#11989 Add wht new

cadawson committed:
Merge branch '2.3_stage' into dawson1198923
